app/src/main/java/org/connectbot/TerminalView.java

```diff
@@ -150,6 +171,282 @@ public TerminalView(Context context, TerminalBridge bridge) {
 
 		// Enable accessibility features if a screen reader is active.
 		new AccessibilityStateTester().execute((Void) null);
+
+		clipboard = (ClipboardManager) context.getSystemService(Context.CLIPBOARD_SERVICE);
+
+		setTextColor(Color.TRANSPARENT);
+		setTypeface(Typeface.MONOSPACE);
+		onFontSizeChanged(bridge.getFontSize());
+
+		if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.HONEYCOMB) {
+			setTextIsSelectable(true);
+
+			initSelectionCallback();
+
+			gestureDetector = new GestureDetector(context, new GestureDetector.SimpleOnGestureListener() {
+				private TerminalBridge bridge = TerminalView.this.bridge;
+				private float totalY = 0;
+
+				@Override
+				public boolean onScroll(MotionEvent e1, MotionEvent e2, float distanceX, float distanceY) {
+					// if releasing then reset total scroll
+					if (e2.getAction() == MotionEvent.ACTION_UP) {
+						totalY = 0;
+					}
+
+					totalY += distanceY;
+					final int moved = (int) (totalY / bridge.charHeight);
+
+					if (moved != 0) {
+						int base = bridge.buffer.getWindowBase();
+						bridge.buffer.setWindowBase(base + moved);
+						totalY = 0;
+
+						refreshTextFromBuffer();
```

Comment:
Doesn't this only really need to be done here if we're already in selection mode? Is there any noticeable performance impact here when scrolling while text is selected?